va/va.go

```diff
@@ -732,23 +727,25 @@ func (va *ValidationAuthorityImpl) PerformValidation(ctx context.Context, req *v
 		logEvent.UsedRSAKEX = record.UsedRSAKEX || logEvent.UsedRSAKEX
 	}
 
-	va.metrics.localValidationTime.With(prometheus.Labels{
-		"type":   string(challenge.Type),
-		"result": string(challenge.Status),
-	}).Observe(localValidationLatency.Seconds())
-	va.metrics.validationTime.With(prometheus.Labels{
-		"type":         string(challenge.Type),
-		"result":       string(challenge.Status),
-		"problem_type": problemType,
-	}).Observe(validationLatency.Seconds())
+	if err != nil {
+		logEvent.InternalError = err.Error()
+		prob = detailedError(err)
+		return bgrpc.ValidationResultToPB(records, filterProblemDetails(prob))
+	}
 
-	va.log.AuditObject("Validation result", logEvent)
+	// Do remote validation. We do this after local validation is complete, to
+	// avoid wasting work when validation will fail anyway. This only returns a
+	// singular problem, because the remote VAs have already audit-logged their
+	// own validation records, and it's not helpful to present multiple large
+	// errors to the end user.
+	prob = va.performRemoteValidation(ctx, req)
```

Comment:
It feels awkward that in the first half of this function we consider `err` the primary source of errors, and derive `prob` from it, but starting with this call we consider `prob` the primary source of errors and return that. I feel like moving the metrics increment and the `prob.Detail` modification into `performRemoteValidation` might help, since that makes the latter half of this function just:

```go
	prob = va.performRemoteValidation(ctx, req)
	return bgrpc.ValidationResultToPB(records, filterProblemDetails(prob))
```

Reply:
Elegant, I like it.